src/os/bluestore/BlueStore.cc

```diff
@@ -7469,7 +7470,11 @@ int BlueStore::_do_read(
       bufferlist& compressed_bl = *p++;
       if (_verify_csum(o, &bptr->get_blob(), 0, compressed_bl,
 		       b2r_it->second.front().logical_offset) < 0) {
-	return -EIO;
+        if (retries) {
+          return _do_read(c, o, offset, length, bl, op_flags, retries - 1);
```

Comment:
Suggest to log here and there with level level small enough to see that by default.

Reply:
Not 100% sure but may be it makes sense to retry in case of zero-filled block only.
And it would be nice to have a comment (referring to a ticket) why we're doing all that stuff here.

Reply:
* _verify_csum already logs to derr if the verification fails - or did you mean to log if a retried read succeeds?

* I *think* that we might get reads that are only partially zero (cf. the rocksdb version of that issue)

* Comments: sure, will add that

Reply:
You're right, pages will be fully zeroed (at least that's what we've seen) but the crc may be over multiple pages.  Comments++!

Reply:
and yet another suggestion - let's make this retry count configurable.